src/System Application/App/User Settings/src/UserSettingsImpl.Codeunit.al

```diff
@@ -131,7 +131,7 @@ codeunit 9175 "User Settings Impl."
         UserSettingsRec."Locale ID" := UserPersonalization."Locale ID";
         UserSettingsRec."Time Zone" := UserPersonalization."Time Zone";
 
-        if CompanyName() <> '' then
+        if (UserPersonalization.Company = '') and (CompanyName() <> '') then
             UserSettingsRec.Company := CopyStr(CompanyName(), 1, 30)
         else
             UserSettingsRec.Company := UserPersonalization.Company;
```

Comment:
We will hit this code if UserPersonalization.Company is blank BUT CompanyName() is not. Wouldn't that make UserSettingsRec.Company blank?

Reply:
If personalization company is blank and company name is not, then we copy from CompanyName which is not blank (since a company name is missing from user personalization), otherwise we use the company already specified in user personalization which of course could be blank (I.E. both company name and user personalization company is blank).